### PR TITLE
214 improvement reimplementation of strava

### LINKED ISSUE #214: [IMPROVEMENT]: reimplementation of strava
### Description

reimplementation of strava

### To do

Oauth connection to strava API into client mobile app.
add strava to areaLoop.
creation of straveService.js containing all strava actions.

### Priority

High

### Project parts

- [X] Server
- [X] Application
- [X] Web
- [ ] Automation

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by STom6**
# New feature
Oauth connection to strava API into client mobile app.
add strava to areaLoop.
creation of straveService.js containing all strava actions.

### CODE REVIEW

File: Application/src/Common/Areas.ts [hunk 1]
```diff
@@ -105,6 +105,48 @@ export let ACTIONS: InfoArea[] = [
         description: "Si krl à plus de 1000 viewers",
         text: "krl_stream__1000",
         subject : "viewers"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à vélo.",
```
Comment (JohanCDev) [inline Application/src/Common/Areas.ts]: Respect french language please
"Si 1000km ont étés faits à vélo"

File: Application/src/Common/Areas.ts [hunk 2]
```diff
@@ -105,6 +105,48 @@ export let ACTIONS: InfoArea[] = [
...
+        text: "",
+        subject : "bikeStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à pied.",
```
Comment (JohanCDev) [inline Application/src/Common/Areas.ts]: Same

File: Application/src/Common/Areas.ts [hunk 3]
```diff
@@ -105,6 +105,48 @@ export let ACTIONS: InfoArea[] = [
...
+        text: "",
+        subject : "runStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à la nage.",
```
Comment (JohanCDev) [inline Application/src/Common/Areas.ts]: Same

File: Application/src/Common/Areas.ts [hunk 4]
```diff
@@ -105,6 +105,48 @@ export let ACTIONS: InfoArea[] = [
...
+        text: "",
+        subject : "swinStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si une nouvelle activité est ajouté.",
```
Comment (JohanCDev) [inline Application/src/Common/Areas.ts]: "AjoutéE"

File: Web/src/Common/Areas.js [hunk 5]
```diff
@@ -103,6 +103,48 @@ export let ACTIONS = [
         description: "Si krl à plus de 1000 viewers",
         text: "krl_stream__1000",
         subject : "viewers"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à vélo.",
```
Comment (JohanCDev) [inline Web/src/Common/Areas.js]: Same

File: Web/src/Common/Areas.js [hunk 6]
```diff
@@ -103,6 +103,48 @@ export let ACTIONS = [
...
+        text: "",
+        subject : "bikeStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à pied.",
```
Comment (JohanCDev) [inline Web/src/Common/Areas.js]: Same

File: Web/src/Common/Areas.js [hunk 7]
```diff
@@ -103,6 +103,48 @@ export let ACTIONS = [
...
+        text: "",
+        subject : "runStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si 1000km on était fait à la nage.",
```
Comment (JohanCDev) [inline Web/src/Common/Areas.js]: Same

File: Web/src/Common/Areas.js [hunk 8]
```diff
@@ -103,6 +103,48 @@ export let ACTIONS = [
...
+        text: "",
+        subject : "swinStats"
+    },
+    {
+        serviceName : "strava",
+        trigger: true,
+        description: "Si une nouvelle activité est ajouté.",
```
Comment (JohanCDev) [inline Web/src/Common/Areas.js]: Same

**Review by JohanCDev — CHANGES_REQUESTED** [review]
French fouls 🙃🔫

**Review by JohanCDev — DISMISSED** [review]
Lgtm

**Review by JohanCDev — APPROVED** [review]
Lgtm

**Review by osvegn — APPROVED** [review]

**Pull request merged**